mobile/src/main/java/org/openhab/habdroid/core/OpenHABVoiceService.java

```diff
@@ -10,120 +10,74 @@
-            mOpenHABTracker = new OpenHABTracker(OpenHABVoiceService.this, getString(R.string.openhab_service_type));
-            mOpenHABTracker.start();
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        String voiceCommand = extractVoiceCommand(intent);
+        if (!voiceCommand.isEmpty()) {
+            try {
+                Connection conn = ConnectionFactory.getConnection(Connection.TYPE_ANY);
+                sendItemCommand("VoiceCommand", voiceCommand, conn, startId);
+            } catch (ConnectionException e) {
+                Log.w(TAG, "Couldn't determine OpenHAB URL", e);
+                showToast(getString(R.string.error_couldnt_determine_openhab_url));
+                stopSelf(startId);
+            }
+        } else {
+            stopSelf(startId);
         }
+        return START_NOT_STICKY;
     }
 
-    /**
-     * Buffers the {@link Intent} to be processed later when openHABBaseUrl has been determined by {@link OpenHABTracker}.
-     *
-     * Usually, the discovery of the openHABBaseUrl is fast enough, so there will be only one intent in the buffer
-     * when the buffer is processed and this service is stopped.
-     * However, it is not guaranteed that this service is only called once before openHABBaseUrl can be discovered.
-     * Therefore all intents are buffered and later this buffer will be processed.
-     *
-     * @param intent The {@link Intent} to be buffered.
-     */
-    private void bufferIntent(Intent intent) {
-        mBufferedIntents.add(intent);
-    }
-
+    @Nullable
     @Override
-    public void onOpenHABTracked(String baseUrl) {
-        Log.d(TAG, "onOpenHABTracked(): " + baseUrl);
-        mOpenHABBaseUrl = baseUrl;
-        while (!mBufferedIntents.isEmpty()) {
-            processVoiceIntent(mBufferedIntents.poll());
-        }
-        Log.d(TAG, "Stopping service for start ID " + getLastStartId());
-        stopSelf(getLastStartId());
+    public IBinder onBind(Intent intent) {
+        return null;
     }
 
     /**
-     * @param message message to show
-     * @param messageType must be Constants.MESSAGES.DIALOG or Constants.MESSAGES.TOAST
-     * @param logLevel not implemented
+     * @param message     message to show
+     * @param messageType must be MessageHandler.TYPE_DIALOG or MessageHandler.TYPE_TOAST
+     * @param logLevel    not implemented
      */
     public void showMessageToUser(String message, int messageType, int logLevel) {
-        if(message == null) {
+        if (message == null) {
             return;
         }
         switch (messageType) {
-            case Constants.MESSAGES.DIALOG:
+            case MessageHandler.TYPE_DIALOG:
                 AlertDialog.Builder builder = new AlertDialog.Builder(OpenHABVoiceService.this);
                 builder.setMessage(message)
                         .setPositiveButton("OK", new DialogInterface.OnClickListener() {
```

Comment:
Hardcoded "OK"

Reply:
Sorry for the late reply! Done, as well as the failing test :)